Merge Ado's navigation to master

ViliamVadocz opened the pull request.

ViliamVadocz committed:
Added sublime files & VSCode workspace to gitignore

ViliamVadocz committed:
Added favicon

ViliamVadocz committed:
Added back the refferences from gitignore.io

ViliamVadocz committed:
Added basic navbar

ViliamVadocz committed:
Added primitive navbar

ViliamVadocz committed:
Made navbar a bit nicer

ViliamVadocz committed:
Merged Ado's changes to my fork

ViliamVadocz committed:
Merge pull request #2 from ViliamVadocz/GalileoSchool-master

Merged Ado's changes with mine

AdrianJakubcik committed:
Changes:
-Reworked the Navigation
- Made it collapsable
- Made it Responsive
- Some style

TODO:
Better style, tweak out the functionality, review.

AdrianJakubcik committed:
Changes:
- <a> was useless so use <li> instead

TODO:
Consider using JQuery instead of lame JS

AdrianJakubcik committed:
Added styles for more media

Added resize event into JavaScript

AdrianJakubcik committed:
Added html comments for educational purposes

AdrianJakubcik committed:
Added Documentation for the JavaScript
(Educational Purposes)

In case of any questions or a weak documentation feel free to contact me

AdrianJakubcik committed:
Added CSS documentation for @media Queries

AdrianJakubcik committed:
Spacing Changes In Documentation

AdrianJakubcik committed:
Some html spacing tweaking

AdrianJakubcik commented:
Btw There is no more need for the include.js in js folder and the navigation.html is not used anymore.

Commit:
Got rid of unnecessary files

AdrianJakubcik committed:
Added Function test that does literally nothing

AdrianJakubcik committed:
Test() now returns null

AdrianJakubcik committed:
Navigation Rework In Progress

AdrianJakubcik committed:
Navigation Bar nearly reworked
Language and Search buttons lack functionality for now
Removed most of Css queries no use for them
Made most of the Navigation bar responsive (Mobile devices only for now)
Added animation using jQuery

AdrianJakubcik committed:
Css tweaking

AdrianJakubcik committed:
TODO: Finish Language Dropdown

AdrianJakubcik committed:
Added Language button dropdown

Added some Css, Javascript, and jQuery

AdrianJakubcik committed:
Fixed Language dropdown hiding bug

DiscreteConcrete commented on js/navigation.js:
```diff
@@ -0,0 +1,73 @@
+/** This is a debug message that tells me wether or not my javascript was loaded, this debug message can be checked in developer tools in chrome under the tab named console
+ * 
+ * Difficulty: Easy
+*/
```
these expanded comments are nicer when you add an extra space before, so that the asterisks are aligned

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,208 @@
+@import url('https://fonts.googleapis.com/css?family=Didact+Gothic');
+
+/* This is the default style that will be used, in-case there are no media queries this is the default style*/
```
in case is two separate words

AdrianJakubcik replied:
Yeah I wasn't sure 'bout that one so I went with the double case. 🤣

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,208 @@
+@import url('https://fonts.googleapis.com/css?family=Didact+Gothic');
+
+/* This is the default style that will be used, in-case there are no media queries this is the default style*/
```
let's be a bit cleaner on formatting here: let's put space before the `*/` in comments

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,208 @@
+
+/* This is the default style that will be used, in-case there are no media queries this is the default style*/
+/* I like to start with the mobile viewport first as my base and then just expand on that code using queries etc.*/
+.nav-desk{
```
let's be a bit stricter on formatting here as well: space before `{`

DiscreteConcrete replied:
what does the "desk" mean here? Desktop? the naming is a bit confusing

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,208 @@
+    }
+
+    #search:hover{
+    background-color:#e9e9e9;
```
indent

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,208 @@
+    }
+
+    #language:hover{
+    background-color: #e7e7e7;
```
indent

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,208 @@
+    display: block;
+}
+
+.nav-desk > ul{
```
I really like how verbose you are in some parts of the codebase - but here there are some non-obvious properties that we are evoking (like `padding-inline-start`) - can we add some more comments?

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,208 @@
+    flex-direction: row;
+    justify-content: flex-start;
+    float: left;
+    z-index: 4;
```
why 4?

AdrianJakubcik replied:
Ohh that was just for a debug purpose, forgot to remove it. Sorry 🤕

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,208 @@
+    background-color: #cccccc !important;
+}
+
+/* This is a query that tells the browser what style to use in case its width is > or = to 740px */
```
it says 670 in the code - so why 740?

AdrianJakubcik replied:
Originally it was 740 but after a little bit of debuging I thought 670 will be better, and forgot to change the comment. Thank you for pointing out 😄  👍

DiscreteConcrete commented on index.html:
```diff
@@ -1,13 +1,71 @@
-	</body>
-</html>
+
+<head>
```
yeh, this is a popular indentation choice - maybe let's add a quick comment that this is deliberate - so those who read this code don't think that we're just lazy with indentation!

DiscreteConcrete commented on js/navigation.js:
```diff
@@ -0,0 +1,73 @@
+ * Difficulty: Easy
+*/
+const NavMenuDesk = document.getElementById('nav-desk'); 
+var NavMenu = document.getElementById('Navigation');
```
why is one const and the other one var?

AdrianJakubcik replied:
My mistake sorry, fixed it.

DiscreteConcrete commented on js/navigation.js:
```diff
@@ -0,0 +1,73 @@
+            return;
+    }
+    else if(window.innerWidth < 670)
+    {
```
we're a bit inconsistent here about where the `{` goes - maybe would be nicer on previous line

DiscreteConcrete commented on js/navigation.js:
```diff
@@ -0,0 +1,73 @@
+function check_resize(){
+    if(window.innerWidth >= 670){
+        if(document.querySelector('.content').classList.contains('blur'))
+            document.querySelector('.content').classList.remove('blur');
```
we're a bit inconsistent here about usage of `{` and `}` - the general rule of thumb in the industry is that if the `if` or `else` and the effect are on the same single line, we don't use braces. however, when it's on the next line(s), then we always use braces.

DiscreteConcrete commented on js/navigation.js:
```diff
@@ -0,0 +1,73 @@
+ */
+function check_resize(){
+    if(window.innerWidth >= 670){
+        if(document.querySelector('.content').classList.contains('blur'))
```
we're also inconsistent about space after `if` - in the industry it is more popular to consistently place a space after the `if`

DiscreteConcrete reviewed (changes requested):
Very nice!! I added comments mostly regarding formatting and code hygiene.
A general style comment would be that in many companies, there is a preference for a maximum line length of 80 characters - that way the reviews are much nicer and you can view the code in a narrower editor.
We could try to use a "linter" like "ES-Lint" or a tool like "Prettier" for this, but maybe later :) 
I will do a second review - a review of functionality later, but it's looking good.

AdrianJakubcik committed:
Added Search Bar

Language selector should be fully functional

Some Javascript and CSS Tweaks

AdrianJakubcik committed:
Requested changes have been made!

AdrianJakubcik committed:
Added Comments For Better Understanding

AdrianJakubcik committed:
Minor jQuery Fixing of Search Bar

AdrianJakubcik committed:
Fixed Search Input border non-responsive

Font-resizing in different desktop viewports

AdrianJakubcik committed:
Merge branch 'master' into Navigation

AdrianJakubcik committed:
Navigation reworked according to the design

I hope it was a final rework

Small tweaking etc. can still be done

AdrianJakubcik committed:
Fixed navigation resetting bug

Fixed desktop viewport cells not being centered

Search bar doesn't overlap logo anymore

AdrianJakubcik committed:
Fixed language divs not being perfect square

AdrianJakubcik committed:
Added meta-tag for viewport

AdrianJakubcik committed:
Merge pull request #1 from AdrianJakubcik/Navigation

Added meta-tag for viewport

ViliamVadocz committed:
Delete workspace file

ViliamVadocz committed:
Add VSCode workspace files to gitignore

AdrianJakubcik committed:
No idea what I did, but it seems to work

AdrianJakubcik committed:
Merge pull request #2 from AdrianJakubcik/Navigation

No idea what I did, but it seems to work

AdrianJakubcik committed:
Css tweaking

AdrianJakubcik committed:
Merge pull request #3 from AdrianJakubcik/Navigation

Css tweaking

AdrianJakubcik committed:
Merge branches 'Navigation' and 'Navigation' of https://github.com/ViliamVadocz/GalileoSchool.github.io into Navigation

AdrianJakubcik committed:
Fixed bug navigation bar disappearing.

AdrianJakubcik committed:
Merge pull request #4 from AdrianJakubcik/Navigation

Navigation

AdrianJakubcik committed:
Fixed uncentered Navigation

AdrianJakubcik committed:
Merge pull request #5 from AdrianJakubcik/Navigation

Fixed uncentered Navigation

AdrianJakubcik committed:
Resolving conflicts to allow merging

AdrianJakubcik committed:
Merge pull request #6 from AdrianJakubcik/Navigation

Resolving conflicts to allow merging

AdrianJakubcik committed:
Fixed decentered, removed li elements

- Some Css tweaking

AdrianJakubcik committed:
Merge branch 'master' into Navigation

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,354 @@
+    display: none;
+    margin: auto;
+    width: 100vw;
+    /* I will be using vw (view-width) units, this is the measure of the browsers content window width */
```
let's put comments ABOVE the lines on which they comment

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,354 @@
+    display: block;
+}
+
+.nav-desk>ul {
```
a more common industry standard is to have spaces around the `>`

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,354 @@
+    /* This tells the browser that I don't want any padding at the start of the line in my whole unordered list, without this our items inside the ul would be shifted towards right due to this padding value not being 0 */
+}
+
+div.nav-phone.lang{
```
let's adopt the convention of space before `{`

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,354 @@
+    justify-content: center;
+}
+
+.nav-phone.language-dropdown-content {
```
this is inconsistent - maybe let's adopt the convention of spaces between classes (before `.`) in CSS blocks?

ViliamVadocz replied:
I think that has a different meaning based on this thread: https://stackoverflow.com/questions/5116288/how-to-select-an-element-with-2-classes

DiscreteConcrete replied:
completely missed that!

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,354 @@
+    display: none;
+    top: 10vw;
+    z-index: 1;
+    /*border-radius: 0.25vw;*/
```
do we need to keep this as a comment?

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,354 @@
+
+.nav-phone .search-input:focus {
+    opacity: 1 !important;
+    /* !important is a special css attribute that overrides any styles that could possibly clash with it (It's like the Principle of Gymnasium compared to the Teacher of Gymnasium */
```
missing ending `)` in comment
also, in English Gymnasium means a workout/sports gym. High school is a better translation :)

ViliamVadocz replied:
also note principle  != principal

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,354 @@
+
+.title-open-nav {
+    color: ghostwhite !important;
+    /*margin-left: 2vw !important;*/
```
do we need to keep this?

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,354 @@
+
+
+/* This is a query that tells the browser what style to use in case its width is > or = to 670px (px stands for pixels) */
+
```
why so many empty lines between them?

DiscreteConcrete replied:
this is still unresolved...

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,354 @@
+}
+
+
+/* This is a query that tells the browser what style to use in case its width is > or = to 670px (px stands for pixels) */
```
let's explain how we got to 670 - because the next person working on it might wonder why such exact number

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,354 @@
+        margin-bottom: 0 !important;
+        position: relative;
+        color: black;
+        /*border-bottom: 2px solid black;*/
```
do we need to keep this?

DiscreteConcrete commented on index.html:
```diff
@@ -1,17 +1,112 @@
-		<p>Work in progress... Check back later!</p>
-	</body>
-</html>
+<!--This Indention is Totally Legal and We Are 100% Certain About This-->
```
why so many capitalised letters? :)

DiscreteConcrete replied:
Also, there tends to be a pedantic convention in the community that there should be a space after `<!--` and before `-->` - and also in CSS after `/*` and before `*/`

DiscreteConcrete commented on index.html:
```diff
@@ -1,17 +1,112 @@
+<head>
+    <!--With this meta tag we are telling the browser what character set we will be using on our web-page (It's crucial for the Slovak diacritic to work, this is because ASCII the default character set doesn't include the weird Slovak symbols like ´ or ô or ˇ etc.) You can read more about this at [https://www.w3schools.com/html/html_charset.asp] -->
+    <meta charset="utf-8">
+    <!-- ADD COMMENT -->
```
?

DiscreteConcrete commented on index.html:
```diff
@@ -1,17 +1,112 @@
+            <div id="nav-bar" class="nav-phone">
+                <div class="left">
+                    <div class="nav-phone search-drop dropdown square">
+                        <div id="search" class="nav-phone search"><img id="menu-img" src="images/icons/search.png">
```
maybe would be nicer if img was on the next line?

DiscreteConcrete commented on index.html:
```diff
@@ -1,17 +1,112 @@
+                        </div>
+                    </div>
+                    <div id="logo-container" class="nav-phone school-logo-container">
+                        <li class="nav-phone logo"><img id="logo" src="images/logo.jpg"></li>
```
maybe would be nicer if img had its own line?

DiscreteConcrete commented on js/jquery.js:
```diff
@@ -0,0 +1,97 @@
+// This tells the browser to execute certain function as soon as it is ready (as soon as the page is fully loaded)
```
`to execute certain function` sounds grammatically awkward - maybe an article would help or maybe it should be plural?

DiscreteConcrete commented on js/jquery.js:
```diff
@@ -0,0 +1,97 @@
+// This tells the browser to execute certain function as soon as it is ready (as soon as the page is fully loaded)
+$(document).ready(function() {
+
+    // Using Id I'm selecting the element with the given Id in this case it is the navigation button
```
i feel some punctuation here would help

DiscreteConcrete commented on js/jquery.js:
```diff
@@ -0,0 +1,97 @@
+
+    // "Installing" Click event listener on the selected element (When clicked the navigation box slides up and down)
+    $(navBtn).click(function() {
+        $('.navbar').slideToggle('fast');
```
there seems to be a lot happening in this function - it's a bit hard to follow!

DiscreteConcrete commented on js/jquery.js:
```diff
@@ -0,0 +1,97 @@
+        function() {
+            $('.language-dropdown-content').fadeToggle(200);
+            return;
+        });
```
would look nicer if the `);` was on the next line

AdrianJakubcik replied:
My formatter always moves it back after the `}`. I'm sorry for that.

DiscreteConcrete replied:
not convinced

DiscreteConcrete commented on js/jquery.js:
```diff
@@ -0,0 +1,97 @@
+            return;
+        });
+
+    $(document).click(function() {
```
this feels a bit dense

AdrianJakubcik committed:
Sk translation fixed

DiscreteConcrete commented on js/jquery.js:
```diff
@@ -0,0 +1,97 @@
+        });
+
+    $(document).click(function() {
+        if (!$('.language-dropdown-content').is(":hover") && !$('.language-dropdown-content').is(":hidden") && !$(langBtn).is(":hover")) {
```
we are a little bit inconsistent in using `'` and `"`

DiscreteConcrete commented on js/jquery.js:
```diff
@@ -0,0 +1,97 @@
+        $(title).removeClass('no-display');
+    };
+
+    //Use the code below for the desktop version (It is the hover version)
```
if we need the code below, maybe we can explain why we commented it out for now?

DiscreteConcrete commented on js/navigation.js:
```diff
@@ -0,0 +1,110 @@
+else
+    var exit_img_src = "images/icons/close.png";
+
+/**
```
maybe we're missing an actual explanation of what this does?

DiscreteConcrete commented on js/navigation.js:
```diff
@@ -0,0 +1,110 @@
+var original_attribute_src = menuBtnImg.getAttribute('src');
+var parsed_dir = Parse_Url(window.location.pathname, 2);
+
+if (parsed_dir == "sk")
```
usually, when the if is multiline, the convention is usually to include the `{` and `}`

DiscreteConcrete commented on js/navigation.js:
```diff
@@ -0,0 +1,110 @@
+ * 
+ * Difficulty: Easy
+ */
+const menuBtn = document.getElementById("nav-btn");
```
use of const is interesting: usually people would either use `var` everywhere (old javascript) or `const` and `let` instead (but not `var`). What is our reason to use `const` and `var`? and how much of the world supports the new javascript with `const` and `let` by now?

AdrianJakubcik replied:
I decided to use `const` for things that are constantly the same no matter what. Like html elements, when I select any html element I know it won't be changing anytime soon, thus I assign it as a `const`. But things that may change throughout the code are assigned as `var` meaning variable. Variable is something that may change because it is vary-able or able to vary. It's totally just my way of dividing constants and non-constants. Btw the support for `const` is around `98.97%`.

DiscreteConcrete replied:
Good! Now what reasons do we have to use `var` over `let`?

DiscreteConcrete commented on js/navigation.js:
```diff
@@ -0,0 +1,110 @@
+/** This is a debug message that tells me wether or not my javascript was loaded, this debug message can be checked in developer tools in chrome under the tab named console
+ * 
+ * Difficulty: Easy
```
do we still need the difficulty indicators?

AdrianJakubcik replied:
I'm leaving them there for the others, if they would like to start with javascript I'm trying to show them what things are easier to do and which are harder to do.

ViliamVadocz replied:
I think they just make the code messier.

AdrianJakubcik replied:
As long as I know the point of a documentation is to include any helpful information that would explain the purpose of a function and/or the way it works. By helpful information I mean anything that can help others understand the function. By telling others how difficult a certain code may be, can give them a way to compare their abilities. Making it easier to choose the most suitable task for them. In our case I do include some unnecessary information but bear in mind I'm not doing it for myself, but for the others.

DiscreteConcrete replied:
I agree with the general point of helping others, but I am sure there is a level where we would be over-commenting (for example if I was to talk about what I had for lunch while coding those parts of code). In this particular case, I feel like we are slightly overstepping the line and it is a bit unnecessary - in other words I don't think there is that much value and giving so many lines of code to something with so little value feels like overcommenting

DiscreteConcrete commented on js/navigation.js:
```diff
@@ -0,0 +1,110 @@
+/** Function that shows and hides our menu on mobile devices
+ * 
+ * Difficulty: Harder
+ * @description What this function does to achieve the effect of hiding and showing our menu after clicking a button on a mobile device, 
```
I feel this description feels a bit grammatically awkward

DiscreteConcrete commented on js/navigation.js:
```diff
@@ -0,0 +1,110 @@
+/** Function that shows and hides our menu on mobile devices
+ * 
+ * Difficulty: Harder
+ * @description What this function does to achieve the effect of hiding and showing our menu after clicking a button on a mobile device, 
```
also, for what reasons do we use the @description here, but not in many other places?

AdrianJakubcik replied:
As I was playing a little bit with Js it seems that Javascript only allows to add description to function, class or any other object that can be called during script execution more than once. Meaning I can't add description to code that is not enclosed and turned into a 'static' object, it's not static but I can't find the right word for it atm. This is the reason why not many things have `@description` but most of my function documentations do include a `@description`.

DiscreteConcrete replied:
fair!

DiscreteConcrete commented on js/navigation.js:
```diff
@@ -0,0 +1,110 @@
+        if (menuBtnImg.getAttribute('src') == exit_img_src) {
+            menuBtnImg.setAttribute('src', original_attribute_src);
+        }
+        if (window.getComputedStyle(NavMenu).display == "none" || window.getComputedStyle(NavMenu).display == "block") {
```
I like this entire function - but maybe we could add a tiny bit more explanation?

DiscreteConcrete commented on js/navigation.js:
```diff
@@ -0,0 +1,110 @@
+    }
+}
+
+function resetMenu() {
```
maybe we can explain what this does?

DiscreteConcrete replied:
I think I am still a bit confused about this function - I understand what it does, but if I change something in the navbar, how will I know I have to edit something here as well? This has a high likelihood of breaking in the future! What can we do about that?

ViliamVadocz replied:
Maybe we could reuse the function found in `jquery.js` which activates on click to toggle the menu visibility which would reduce the amount of code that needs to be updated if anything changes.

AdrianJakubcik committed:
Changes requested on pull request

AdrianJakubcik committed:
Merge branch 'Navigation' of https://github.com/ViliamVadocz/GalileoSchool.github.io into Navigation

AdrianJakubcik committed:
Attempt to clean the Css file

DiscreteConcrete commented on css/nav_style.css:
```diff
@@ -0,0 +1,362 @@
+
+.nav-phone .search-input:focus {
+    opacity: 1 !important;
+    /* !important is a special css attribute that overrides any styles that could possibly clash with it (It's like the Principle of High School compared to the Teacher of High School) */
```
move comment above what interests us

AdrianJakubcik committed:
Requested changes in css and jquery files

AdrianJakubcik committed:
Merge branch 'master' into Navigation

ViliamVadocz commented:
Would be nice if we resolved already solved comments so it is easier to see what needs addressing.

AdrianJakubcik committed:
Fixed javascript bug, little Css tweaking

ViliamVadocz committed:
Shortened comments and put spaces around >

ViliamVadocz committed:
Cleaned up comments

ViliamVadocz committed:
Enforced lowercase class names

ViliamVadocz committed:
Minor colour correction and class rename

ViliamVadocz committed:
Cleaned up js files

ViliamVadocz committed:
Accidentally disabled resetMenu while testing

ViliamVadocz committed:
Updated SK version
